<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<html lang="en">
  <head>
    <meta charset="utf-8">
<!--
    <link rel="stylesheet" type="text/css" href="http://epigenomegateway.wustl.edu/browser/style.css" />
    <script type="text/javascript" src="http://epigenomegateway.wustl.edu/browser/js/base.js"></script>
    <script type="text/javascript" src="http://epigenomegateway.wustl.edu/browser/js/personality.js"></script>
    <script type="text/javascript" src="http://epigenomegateway.wustl.edu/browser/js/embed.js"></script>
-->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
    
    <link href="./report/jquery.treetable.css" rel="stylesheet" type="text/css" />
    <link href="./report/jquery.treetable.theme.default.css" rel="stylesheet" />
    <script src="./report/jquery.treetable.js"></script>

    <script type="text/javascript">
      $( function(){
      	$("#filetable").treetable({ 
      		expandable: true,
      		onNodeCollapse: function() {
      		},
      		onNodeExpand: function() {
      		}
      	});
      });
    </script>
  </head>
  <body>
<div id='title'><b>Title: MITF_ANCG_Output</b><br><p>Report generated at Tue Feb  4 03:52:35 CST 2020 <br></p></div><div id='cmd_line_args'><b>Command line arguments</b><br><p>-out_dir MITF_ANCG_Output -type -se -nth 8 -extsize_macs2 200 -species hg38 -fastq2:1 /gpfs/gpfs2/home/jloupe/Flowcells/HLKL7DMXX/trim_fastq/30M.GSLv5-8_i7_165.merge.fastq.gz -fastq1:1 /gpfs/gpfs2/home/jloupe/Flowcells/HLKL7DMXX/trim_fastq/30M.GSLv5-8_i7_160.merge.fastq.gz -ctl_fastq1:1 /gpfs/gpfs2/home/jloupe/Flowcells/HLKL7DMXX/trim_fastq/30M.GSLv5-8_i7_170.merge.fastq.gz -ctl_fastq2:1 /gpfs/gpfs2/home/jloupe/Flowcells/HLKL7DMXX/trim_fastq/30M.GSLv5-8_i7_175.merge.fastq.gz</p></div><div id='dirs_and_files'><b>Directories and files</b><table id='filetable'> <caption><a href='#' onclick="jQuery('#filetable').treetable('expandAll'); return false;">Expand all</a> &nbsp&nbsp<a href='#' onclick="jQuery('#filetable').treetable('collapseAll'); return false;">Collapse all</a></caption> <thead><tr><th>Files</th><th>Path</th></tr></thead> <tbody><tr data-tt-id='Raw reads'><td> Raw reads </td><td><br></td></tr><tr data-tt-id='Raw reads/Replicate 1' data-tt-parent-id='Raw reads'><td> Replicate 1 </td><td><br></td></tr><tr data-tt-id='Raw reads/Replicate 1/Fastq (29M)' data-tt-parent-id='Raw reads/Replicate 1'><td> Fastq (29M) </td><td>/gpfs/gpfs2/home/jloupe/Flowcells/HLKL7DMXX/trim_fastq/30M.GSLv5-8_i7_160.merge.fastq.gz<br></td></tr><tr data-tt-id='Raw reads/Replicate 2' data-tt-parent-id='Raw reads'><td> Replicate 2 </td><td><br></td></tr><tr data-tt-id='Raw reads/Replicate 2/Fastq (30M)' data-tt-parent-id='Raw reads/Replicate 2'><td> Fastq (30M) </td><td>/gpfs/gpfs2/home/jloupe/Flowcells/HLKL7DMXX/trim_fastq/30M.GSLv5-8_i7_165.merge.fastq.gz<br></td></tr><tr data-tt-id='Raw reads/Control 1' data-tt-parent-id='Raw reads'><td> Control 1 </td><td><br></td></tr><tr data-tt-id='Raw reads/Control 1/Fastq (26M)' data-tt-parent-id='Raw reads/Control 1'><td> Fastq (26M) </td><td>/gpfs/gpfs2/home/jloupe/Flowcells/HLKL7DMXX/trim_fastq/30M.GSLv5-8_i7_170.merge.fastq.gz<br></td></tr><tr data-tt-id='Raw reads/Control 2' data-tt-parent-id='Raw reads'><td> Control 2 </td><td><br></td></tr><tr data-tt-id='Raw reads/Control 2/Fastq (30M)' data-tt-parent-id='Raw reads/Control 2'><td> Fastq (30M) </td><td>/gpfs/gpfs2/home/jloupe/Flowcells/HLKL7DMXX/trim_fastq/30M.GSLv5-8_i7_175.merge.fastq.gz<br></td></tr><tr data-tt-id='Alignment'><td> Alignment </td><td><br></td></tr><tr data-tt-id='Alignment/Replicate 1' data-tt-parent-id='Alignment'><td> Replicate 1 </td><td><br></td></tr><tr data-tt-id='Alignment/Replicate 1/Bam (24M)' data-tt-parent-id='Alignment/Replicate 1'><td> Bam (24M) </td><td><a href='./align/rep1/30M.GSLv5-8_i7_160.merge.bam' target='_blank'>./align/rep1/30M.GSLv5-8_i7_160.merge.bam</a><br></td></tr><tr data-tt-id='Alignment/Replicate 1/Filtered & deduped bam (16M)' data-tt-parent-id='Alignment/Replicate 1'><td> Filtered & deduped bam (16M) </td><td><a href='./align/rep1/30M.GSLv5-8_i7_160.merge.nodup.bam' target='_blank'>./align/rep1/30M.GSLv5-8_i7_160.merge.nodup.bam</a><br></td></tr><tr data-tt-id='Alignment/Replicate 1/Tag-align' data-tt-parent-id='Alignment/Replicate 1'><td> Tag-align </td><td><a href='./align/rep1/30M.GSLv5-8_i7_160.merge.nodup.tagAlign.gz' target='_blank'>./align/rep1/30M.GSLv5-8_i7_160.merge.nodup.tagAlign.gz</a><br></td></tr><tr data-tt-id='Alignment/Replicate 2' data-tt-parent-id='Alignment'><td> Replicate 2 </td><td><br></td></tr><tr data-tt-id='Alignment/Replicate 2/Bam (25M)' data-tt-parent-id='Alignment/Replicate 2'><td> Bam (25M) </td><td><a href='./align/rep2/30M.GSLv5-8_i7_165.merge.bam' target='_blank'>./align/rep2/30M.GSLv5-8_i7_165.merge.bam</a><br></td></tr><tr data-tt-id='Alignment/Replicate 2/Filtered & deduped bam (18M)' data-tt-parent-id='Alignment/Replicate 2'><td> Filtered & deduped bam (18M) </td><td><a href='./align/rep2/30M.GSLv5-8_i7_165.merge.nodup.bam' target='_blank'>./align/rep2/30M.GSLv5-8_i7_165.merge.nodup.bam</a><br></td></tr><tr data-tt-id='Alignment/Replicate 2/Tag-align' data-tt-parent-id='Alignment/Replicate 2'><td> Tag-align </td><td><a href='./align/rep2/30M.GSLv5-8_i7_165.merge.nodup.tagAlign.gz' target='_blank'>./align/rep2/30M.GSLv5-8_i7_165.merge.nodup.tagAlign.gz</a><br></td></tr><tr data-tt-id='Alignment/Control 1' data-tt-parent-id='Alignment'><td> Control 1 </td><td><br></td></tr><tr data-tt-id='Alignment/Control 1/Bam (25M)' data-tt-parent-id='Alignment/Control 1'><td> Bam (25M) </td><td><a href='./align/ctl1/30M.GSLv5-8_i7_170.merge.bam' target='_blank'>./align/ctl1/30M.GSLv5-8_i7_170.merge.bam</a><br></td></tr><tr data-tt-id='Alignment/Control 1/Filtered & deduped bam (21M)' data-tt-parent-id='Alignment/Control 1'><td> Filtered & deduped bam (21M) </td><td><a href='./align/ctl1/30M.GSLv5-8_i7_170.merge.nodup.bam' target='_blank'>./align/ctl1/30M.GSLv5-8_i7_170.merge.nodup.bam</a><br></td></tr><tr data-tt-id='Alignment/Control 1/Tag-align' data-tt-parent-id='Alignment/Control 1'><td> Tag-align </td><td><a href='./align/ctl1/30M.GSLv5-8_i7_170.merge.nodup.tagAlign.gz' target='_blank'>./align/ctl1/30M.GSLv5-8_i7_170.merge.nodup.tagAlign.gz</a><br></td></tr><tr data-tt-id='Alignment/Control 2' data-tt-parent-id='Alignment'><td> Control 2 </td><td><br></td></tr><tr data-tt-id='Alignment/Control 2/Bam (29M)' data-tt-parent-id='Alignment/Control 2'><td> Bam (29M) </td><td><a href='./align/ctl2/30M.GSLv5-8_i7_175.merge.bam' target='_blank'>./align/ctl2/30M.GSLv5-8_i7_175.merge.bam</a><br></td></tr><tr data-tt-id='Alignment/Control 2/Filtered & deduped bam (24M)' data-tt-parent-id='Alignment/Control 2'><td> Filtered & deduped bam (24M) </td><td><a href='./align/ctl2/30M.GSLv5-8_i7_175.merge.nodup.bam' target='_blank'>./align/ctl2/30M.GSLv5-8_i7_175.merge.nodup.bam</a><br></td></tr><tr data-tt-id='Alignment/Control 2/Tag-align' data-tt-parent-id='Alignment/Control 2'><td> Tag-align </td><td><a href='./align/ctl2/30M.GSLv5-8_i7_175.merge.nodup.tagAlign.gz' target='_blank'>./align/ctl2/30M.GSLv5-8_i7_175.merge.nodup.tagAlign.gz</a><br></td></tr><tr data-tt-id='Alignment/Pooled replicate' data-tt-parent-id='Alignment'><td> Pooled replicate </td><td><br></td></tr><tr data-tt-id='Alignment/Pooled replicate/Tag-align' data-tt-parent-id='Alignment/Pooled replicate'><td> Tag-align </td><td><a href='./align/pooled_rep/30M.GSLv5-8_i7_160.merge.nodup_30M.GSLv5-8_i7_165.merge.nodup.tagAlign.gz' target='_blank'>./align/pooled_rep/30M.GSLv5-8_i7_160.merge.nodup_30M.GSLv5-8_i7_165.merge.nodup.tagAlign.gz</a><br></td></tr><tr data-tt-id='Alignment/Pooled control' data-tt-parent-id='Alignment'><td> Pooled control </td><td><br></td></tr><tr data-tt-id='Alignment/Pooled control/Tag-align' data-tt-parent-id='Alignment/Pooled control'><td> Tag-align </td><td><a href='./align/pooled_ctl/30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.gz' target='_blank'>./align/pooled_ctl/30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.gz</a><br></td></tr><tr data-tt-id='Alignment/Pseudo-replicates' data-tt-parent-id='Alignment'><td> Pseudo-replicates </td><td><br></td></tr><tr data-tt-id='Alignment/Pseudo-replicates/Replicate 1' data-tt-parent-id='Alignment/Pseudo-replicates'><td> Replicate 1 </td><td><br></td></tr><tr data-tt-id='Alignment/Pseudo-replicates/Replicate 1/Pseudo-replicate 1' data-tt-parent-id='Alignment/Pseudo-replicates/Replicate 1'><td> Pseudo-replicate 1 </td><td><br></td></tr><tr data-tt-id='Alignment/Pseudo-replicates/Replicate 1/Pseudo-replicate 1/Tag-align' data-tt-parent-id='Alignment/Pseudo-replicates/Replicate 1/Pseudo-replicate 1'><td> Tag-align </td><td><a href='./align/pseudo_reps/rep1/pr1/30M.GSLv5-8_i7_160.merge.nodup.pr1.tagAlign.gz' target='_blank'>./align/pseudo_reps/rep1/pr1/30M.GSLv5-8_i7_160.merge.nodup.pr1.tagAlign.gz</a><br></td></tr><tr data-tt-id='Alignment/Pseudo-replicates/Replicate 1/Pseudo-replicate 2' data-tt-parent-id='Alignment/Pseudo-replicates/Replicate 1'><td> Pseudo-replicate 2 </td><td><br></td></tr><tr data-tt-id='Alignment/Pseudo-replicates/Replicate 1/Pseudo-replicate 2/Tag-align' data-tt-parent-id='Alignment/Pseudo-replicates/Replicate 1/Pseudo-replicate 2'><td> Tag-align </td><td><a href='./align/pseudo_reps/rep1/pr2/30M.GSLv5-8_i7_160.merge.nodup.pr2.tagAlign.gz' target='_blank'>./align/pseudo_reps/rep1/pr2/30M.GSLv5-8_i7_160.merge.nodup.pr2.tagAlign.gz</a><br></td></tr><tr data-tt-id='Alignment/Pseudo-replicates/Replicate 2' data-tt-parent-id='Alignment/Pseudo-replicates'><td> Replicate 2 </td><td><br></td></tr><tr data-tt-id='Alignment/Pseudo-replicates/Replicate 2/Pseudo-replicate 1' data-tt-parent-id='Alignment/Pseudo-replicates/Replicate 2'><td> Pseudo-replicate 1 </td><td><br></td></tr><tr data-tt-id='Alignment/Pseudo-replicates/Replicate 2/Pseudo-replicate 1/Tag-align' data-tt-parent-id='Alignment/Pseudo-replicates/Replicate 2/Pseudo-replicate 1'><td> Tag-align </td><td><a href='./align/pseudo_reps/rep2/pr1/30M.GSLv5-8_i7_165.merge.nodup.pr1.tagAlign.gz' target='_blank'>./align/pseudo_reps/rep2/pr1/30M.GSLv5-8_i7_165.merge.nodup.pr1.tagAlign.gz</a><br></td></tr><tr data-tt-id='Alignment/Pseudo-replicates/Replicate 2/Pseudo-replicate 2' data-tt-parent-id='Alignment/Pseudo-replicates/Replicate 2'><td> Pseudo-replicate 2 </td><td><br></td></tr><tr data-tt-id='Alignment/Pseudo-replicates/Replicate 2/Pseudo-replicate 2/Tag-align' data-tt-parent-id='Alignment/Pseudo-replicates/Replicate 2/Pseudo-replicate 2'><td> Tag-align </td><td><a href='./align/pseudo_reps/rep2/pr2/30M.GSLv5-8_i7_165.merge.nodup.pr2.tagAlign.gz' target='_blank'>./align/pseudo_reps/rep2/pr2/30M.GSLv5-8_i7_165.merge.nodup.pr2.tagAlign.gz</a><br></td></tr><tr data-tt-id='Alignment/Pooled pseudo-replicates' data-tt-parent-id='Alignment'><td> Pooled pseudo-replicates </td><td><br></td></tr><tr data-tt-id='Alignment/Pooled pseudo-replicates/Pooled pseudo-replicate 1' data-tt-parent-id='Alignment/Pooled pseudo-replicates'><td> Pooled pseudo-replicate 1 </td><td><br></td></tr><tr data-tt-id='Alignment/Pooled pseudo-replicates/Pooled pseudo-replicate 1/Tag-align' data-tt-parent-id='Alignment/Pooled pseudo-replicates/Pooled pseudo-replicate 1'><td> Tag-align </td><td><a href='./align/pooled_pseudo_reps/ppr1/30M.GSLv5-8_i7_160.merge.nodup.pr1_30M.GSLv5-8_i7_165.merge.nodup.pr1.tagAlign.gz' target='_blank'>./align/pooled_pseudo_reps/ppr1/30M.GSLv5-8_i7_160.merge.nodup.pr1_30M.GSLv5-8_i7_165.merge.nodup.pr1.tagAlign.gz</a><br></td></tr><tr data-tt-id='Alignment/Pooled pseudo-replicates/Pooled pseudo-replicate 2' data-tt-parent-id='Alignment/Pooled pseudo-replicates'><td> Pooled pseudo-replicate 2 </td><td><br></td></tr><tr data-tt-id='Alignment/Pooled pseudo-replicates/Pooled pseudo-replicate 2/Tag-align' data-tt-parent-id='Alignment/Pooled pseudo-replicates/Pooled pseudo-replicate 2'><td> Tag-align </td><td><a href='./align/pooled_pseudo_reps/ppr2/30M.GSLv5-8_i7_160.merge.nodup.pr2_30M.GSLv5-8_i7_165.merge.nodup.pr2.tagAlign.gz' target='_blank'>./align/pooled_pseudo_reps/ppr2/30M.GSLv5-8_i7_160.merge.nodup.pr2_30M.GSLv5-8_i7_165.merge.nodup.pr2.tagAlign.gz</a><br></td></tr><tr data-tt-id='Signal tracks'><td> Signal tracks </td><td><br></td></tr><tr data-tt-id='Signal tracks/MACS2' data-tt-parent-id='Signal tracks'><td> MACS2 </td><td><br></td></tr><tr data-tt-id='Signal tracks/MACS2/Replicate 1' data-tt-parent-id='Signal tracks/MACS2'><td> Replicate 1 </td><td><br></td></tr><tr data-tt-id='Signal tracks/MACS2/Replicate 1/P-value' data-tt-parent-id='Signal tracks/MACS2/Replicate 1'><td> P-value </td><td><a href='./signal/macs2/rep1/30M.GSLv5-8_i7_160.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup.tagAlign.pval.signal.bw' target='_blank'>./signal/macs2/rep1/30M.GSLv5-8_i7_160.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup.tagAlign.pval.signal.bw</a><br></td></tr><tr data-tt-id='Signal tracks/MACS2/Replicate 1/Fold enrichment' data-tt-parent-id='Signal tracks/MACS2/Replicate 1'><td> Fold enrichment </td><td><a href='./signal/macs2/rep1/30M.GSLv5-8_i7_160.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup.tagAlign.fc.signal.bw' target='_blank'>./signal/macs2/rep1/30M.GSLv5-8_i7_160.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup.tagAlign.fc.signal.bw</a><br></td></tr><tr data-tt-id='Signal tracks/MACS2/Replicate 2' data-tt-parent-id='Signal tracks/MACS2'><td> Replicate 2 </td><td><br></td></tr><tr data-tt-id='Signal tracks/MACS2/Replicate 2/P-value' data-tt-parent-id='Signal tracks/MACS2/Replicate 2'><td> P-value </td><td><a href='./signal/macs2/rep2/30M.GSLv5-8_i7_165.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.pval.signal.bw' target='_blank'>./signal/macs2/rep2/30M.GSLv5-8_i7_165.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.pval.signal.bw</a><br></td></tr><tr data-tt-id='Signal tracks/MACS2/Replicate 2/Fold enrichment' data-tt-parent-id='Signal tracks/MACS2/Replicate 2'><td> Fold enrichment </td><td><a href='./signal/macs2/rep2/30M.GSLv5-8_i7_165.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.fc.signal.bw' target='_blank'>./signal/macs2/rep2/30M.GSLv5-8_i7_165.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.fc.signal.bw</a><br></td></tr><tr data-tt-id='Signal tracks/MACS2/Pooled replicate' data-tt-parent-id='Signal tracks/MACS2'><td> Pooled replicate </td><td><br></td></tr><tr data-tt-id='Signal tracks/MACS2/Pooled replicate/P-value' data-tt-parent-id='Signal tracks/MACS2/Pooled replicate'><td> P-value </td><td><a href='./signal/macs2/pooled_rep/30M.GSLv5-8_i7_160.merge.nodup_30M.GSLv5-8_i7_165.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.pval.signal.bw' target='_blank'>./signal/macs2/pooled_rep/30M.GSLv5-8_i7_160.merge.nodup_30M.GSLv5-8_i7_165.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.pval.signal.bw</a><br></td></tr><tr data-tt-id='Signal tracks/MACS2/Pooled replicate/Fold enrichment' data-tt-parent-id='Signal tracks/MACS2/Pooled replicate'><td> Fold enrichment </td><td><a href='./signal/macs2/pooled_rep/30M.GSLv5-8_i7_160.merge.nodup_30M.GSLv5-8_i7_165.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.fc.signal.bw' target='_blank'>./signal/macs2/pooled_rep/30M.GSLv5-8_i7_160.merge.nodup_30M.GSLv5-8_i7_165.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.fc.signal.bw</a><br></td></tr><tr data-tt-id='Peaks'><td> Peaks </td><td><br></td></tr><tr data-tt-id='Peaks/SPP' data-tt-parent-id='Peaks'><td> SPP </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Pooled Replicate' data-tt-parent-id='Peaks/SPP'><td> Pooled Replicate </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Pooled Replicate/Region peak' data-tt-parent-id='Peaks/SPP/Pooled Replicate'><td> Region peak </td><td><a href='./peak/spp/pooled_rep/30M.GSLv5-8_i7_160.merge.nodup_30M.GSLv5-8_i7_165.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.regionPeak.gz' target='_blank'>./peak/spp/pooled_rep/30M.GSLv5-8_i7_160.merge.nodup_30M.GSLv5-8_i7_165.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.regionPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/Replicate 1' data-tt-parent-id='Peaks/SPP'><td> Replicate 1 </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Replicate 1/Region peak' data-tt-parent-id='Peaks/SPP/Replicate 1'><td> Region peak </td><td><a href='./peak/spp/rep1/30M.GSLv5-8_i7_160.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup.tagAlign.regionPeak.gz' target='_blank'>./peak/spp/rep1/30M.GSLv5-8_i7_160.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup.tagAlign.regionPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/Replicate 2' data-tt-parent-id='Peaks/SPP'><td> Replicate 2 </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Replicate 2/Region peak' data-tt-parent-id='Peaks/SPP/Replicate 2'><td> Region peak </td><td><a href='./peak/spp/rep2/30M.GSLv5-8_i7_165.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.regionPeak.gz' target='_blank'>./peak/spp/rep2/30M.GSLv5-8_i7_165.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.regionPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/Pseudo-replicates' data-tt-parent-id='Peaks/SPP'><td> Pseudo-replicates </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Pseudo-replicates/Replicate 1' data-tt-parent-id='Peaks/SPP/Pseudo-replicates'><td> Replicate 1 </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Pseudo-replicates/Replicate 1/Pseudo-replicate 1' data-tt-parent-id='Peaks/SPP/Pseudo-replicates/Replicate 1'><td> Pseudo-replicate 1 </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Pseudo-replicates/Replicate 1/Pseudo-replicate 1/Region peak' data-tt-parent-id='Peaks/SPP/Pseudo-replicates/Replicate 1/Pseudo-replicate 1'><td> Region peak </td><td><a href='./peak/spp/pseudo_reps/rep1/pr1/30M.GSLv5-8_i7_160.merge.nodup.pr1.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup.tagAlign.regionPeak.gz' target='_blank'>./peak/spp/pseudo_reps/rep1/pr1/30M.GSLv5-8_i7_160.merge.nodup.pr1.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup.tagAlign.regionPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/Pseudo-replicates/Replicate 1/Pseudo-replicate 2' data-tt-parent-id='Peaks/SPP/Pseudo-replicates/Replicate 1'><td> Pseudo-replicate 2 </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Pseudo-replicates/Replicate 1/Pseudo-replicate 2/Region peak' data-tt-parent-id='Peaks/SPP/Pseudo-replicates/Replicate 1/Pseudo-replicate 2'><td> Region peak </td><td><a href='./peak/spp/pseudo_reps/rep1/pr2/30M.GSLv5-8_i7_160.merge.nodup.pr2.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup.tagAlign.regionPeak.gz' target='_blank'>./peak/spp/pseudo_reps/rep1/pr2/30M.GSLv5-8_i7_160.merge.nodup.pr2.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup.tagAlign.regionPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/Pseudo-replicates/Replicate 2' data-tt-parent-id='Peaks/SPP/Pseudo-replicates'><td> Replicate 2 </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Pseudo-replicates/Replicate 2/Pseudo-replicate 1' data-tt-parent-id='Peaks/SPP/Pseudo-replicates/Replicate 2'><td> Pseudo-replicate 1 </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Pseudo-replicates/Replicate 2/Pseudo-replicate 1/Region peak' data-tt-parent-id='Peaks/SPP/Pseudo-replicates/Replicate 2/Pseudo-replicate 1'><td> Region peak </td><td><a href='./peak/spp/pseudo_reps/rep2/pr1/30M.GSLv5-8_i7_165.merge.nodup.pr1.tagAlign_x_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.regionPeak.gz' target='_blank'>./peak/spp/pseudo_reps/rep2/pr1/30M.GSLv5-8_i7_165.merge.nodup.pr1.tagAlign_x_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.regionPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/Pseudo-replicates/Replicate 2/Pseudo-replicate 2' data-tt-parent-id='Peaks/SPP/Pseudo-replicates/Replicate 2'><td> Pseudo-replicate 2 </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Pseudo-replicates/Replicate 2/Pseudo-replicate 2/Region peak' data-tt-parent-id='Peaks/SPP/Pseudo-replicates/Replicate 2/Pseudo-replicate 2'><td> Region peak </td><td><a href='./peak/spp/pseudo_reps/rep2/pr2/30M.GSLv5-8_i7_165.merge.nodup.pr2.tagAlign_x_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.regionPeak.gz' target='_blank'>./peak/spp/pseudo_reps/rep2/pr2/30M.GSLv5-8_i7_165.merge.nodup.pr2.tagAlign_x_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.regionPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/Pooled pseudo-replicates' data-tt-parent-id='Peaks/SPP'><td> Pooled pseudo-replicates </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Pooled pseudo-replicates/Pooled pseudo-replicate 1' data-tt-parent-id='Peaks/SPP/Pooled pseudo-replicates'><td> Pooled pseudo-replicate 1 </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Pooled pseudo-replicates/Pooled pseudo-replicate 1/Region peak' data-tt-parent-id='Peaks/SPP/Pooled pseudo-replicates/Pooled pseudo-replicate 1'><td> Region peak </td><td><a href='./peak/spp/pooled_pseudo_reps/ppr1/30M.GSLv5-8_i7_160.merge.nodup.pr1_30M.GSLv5-8_i7_165.merge.nodup.pr1.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.regionPeak.gz' target='_blank'>./peak/spp/pooled_pseudo_reps/ppr1/30M.GSLv5-8_i7_160.merge.nodup.pr1_30M.GSLv5-8_i7_165.merge.nodup.pr1.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.regionPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/Pooled pseudo-replicates/Pooled pseudo-replicate 2' data-tt-parent-id='Peaks/SPP/Pooled pseudo-replicates'><td> Pooled pseudo-replicate 2 </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Pooled pseudo-replicates/Pooled pseudo-replicate 2/Region peak' data-tt-parent-id='Peaks/SPP/Pooled pseudo-replicates/Pooled pseudo-replicate 2'><td> Region peak </td><td><a href='./peak/spp/pooled_pseudo_reps/ppr2/30M.GSLv5-8_i7_160.merge.nodup.pr2_30M.GSLv5-8_i7_165.merge.nodup.pr2.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.regionPeak.gz' target='_blank'>./peak/spp/pooled_pseudo_reps/ppr2/30M.GSLv5-8_i7_160.merge.nodup.pr2_30M.GSLv5-8_i7_165.merge.nodup.pr2.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.regionPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/Naive overlap' data-tt-parent-id='Peaks/SPP'><td> Naive overlap </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Naive overlap/True replicates' data-tt-parent-id='Peaks/SPP/Naive overlap'><td> True replicates </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Naive overlap/True replicates/Rep. 1 vs. Rep. 2' data-tt-parent-id='Peaks/SPP/Naive overlap/True replicates'><td> Rep. 1 vs. Rep. 2 </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Naive overlap/True replicates/Rep. 1 vs. Rep. 2/Overlapping peak' data-tt-parent-id='Peaks/SPP/Naive overlap/True replicates/Rep. 1 vs. Rep. 2'><td> Overlapping peak </td><td><a href='./peak/spp/overlap/true_reps/rep1-rep2/MITF_ANCG_Output_rep1-rep2.naive_overlap.filt.regionPeak.gz' target='_blank'>./peak/spp/overlap/true_reps/rep1-rep2/MITF_ANCG_Output_rep1-rep2.naive_overlap.filt.regionPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/Naive overlap/Pseudo-replicates' data-tt-parent-id='Peaks/SPP/Naive overlap'><td> Pseudo-replicates </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Naive overlap/Pseudo-replicates/Replicate 1' data-tt-parent-id='Peaks/SPP/Naive overlap/Pseudo-replicates'><td> Replicate 1 </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Naive overlap/Pseudo-replicates/Replicate 1/Overlapping peak' data-tt-parent-id='Peaks/SPP/Naive overlap/Pseudo-replicates/Replicate 1'><td> Overlapping peak </td><td><a href='./peak/spp/overlap/pseudo_reps/rep1/MITF_ANCG_Output_rep1-pr.naive_overlap.filt.regionPeak.gz' target='_blank'>./peak/spp/overlap/pseudo_reps/rep1/MITF_ANCG_Output_rep1-pr.naive_overlap.filt.regionPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/Naive overlap/Pseudo-replicates/Replicate 2' data-tt-parent-id='Peaks/SPP/Naive overlap/Pseudo-replicates'><td> Replicate 2 </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Naive overlap/Pseudo-replicates/Replicate 2/Overlapping peak' data-tt-parent-id='Peaks/SPP/Naive overlap/Pseudo-replicates/Replicate 2'><td> Overlapping peak </td><td><a href='./peak/spp/overlap/pseudo_reps/rep2/MITF_ANCG_Output_rep2-pr.naive_overlap.filt.regionPeak.gz' target='_blank'>./peak/spp/overlap/pseudo_reps/rep2/MITF_ANCG_Output_rep2-pr.naive_overlap.filt.regionPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/Naive overlap/Pooled pseudo-replicates' data-tt-parent-id='Peaks/SPP/Naive overlap'><td> Pooled pseudo-replicates </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Naive overlap/Pooled pseudo-replicates/Overlapping peak' data-tt-parent-id='Peaks/SPP/Naive overlap/Pooled pseudo-replicates'><td> Overlapping peak </td><td><a href='./peak/spp/overlap/pooled_pseudo_reps/MITF_ANCG_Output_ppr.naive_overlap.filt.regionPeak.gz' target='_blank'>./peak/spp/overlap/pooled_pseudo_reps/MITF_ANCG_Output_ppr.naive_overlap.filt.regionPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/Naive overlap/Optimal set' data-tt-parent-id='Peaks/SPP/Naive overlap'><td> Optimal set </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Naive overlap/Optimal set/Overlapping peak' data-tt-parent-id='Peaks/SPP/Naive overlap/Optimal set'><td> Overlapping peak </td><td><a href='./peak/spp/overlap/optimal_set/MITF_ANCG_Output_ppr.naive_overlap.filt.regionPeak.gz' target='_blank'>./peak/spp/overlap/optimal_set/MITF_ANCG_Output_ppr.naive_overlap.filt.regionPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/Naive overlap/Conservative set' data-tt-parent-id='Peaks/SPP/Naive overlap'><td> Conservative set </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/Naive overlap/Conservative set/Overlapping peak' data-tt-parent-id='Peaks/SPP/Naive overlap/Conservative set'><td> Overlapping peak </td><td><a href='./peak/spp/overlap/conservative_set/MITF_ANCG_Output_rep1-rep2.naive_overlap.filt.regionPeak.gz' target='_blank'>./peak/spp/overlap/conservative_set/MITF_ANCG_Output_rep1-rep2.naive_overlap.filt.regionPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/IDR' data-tt-parent-id='Peaks/SPP'><td> IDR </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/IDR/True replicates' data-tt-parent-id='Peaks/SPP/IDR'><td> True replicates </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/IDR/True replicates/Rep. 1 vs. Rep. 2' data-tt-parent-id='Peaks/SPP/IDR/True replicates'><td> Rep. 1 vs. Rep. 2 </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/IDR/True replicates/Rep. 1 vs. Rep. 2/IDR peak' data-tt-parent-id='Peaks/SPP/IDR/True replicates/Rep. 1 vs. Rep. 2'><td> IDR peak </td><td><a href='./peak/spp/idr/true_reps/rep1-rep2/MITF_ANCG_Output_rep1-rep2.IDR0.05.filt.narrowPeak.gz' target='_blank'>./peak/spp/idr/true_reps/rep1-rep2/MITF_ANCG_Output_rep1-rep2.IDR0.05.filt.narrowPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/IDR/Pseudo-replicates' data-tt-parent-id='Peaks/SPP/IDR'><td> Pseudo-replicates </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/IDR/Pseudo-replicates/Replicate 1' data-tt-parent-id='Peaks/SPP/IDR/Pseudo-replicates'><td> Replicate 1 </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/IDR/Pseudo-replicates/Replicate 1/IDR peak' data-tt-parent-id='Peaks/SPP/IDR/Pseudo-replicates/Replicate 1'><td> IDR peak </td><td><a href='./peak/spp/idr/pseudo_reps/rep1/MITF_ANCG_Output_rep1-pr.IDR0.05.filt.narrowPeak.gz' target='_blank'>./peak/spp/idr/pseudo_reps/rep1/MITF_ANCG_Output_rep1-pr.IDR0.05.filt.narrowPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/IDR/Pseudo-replicates/Replicate 2' data-tt-parent-id='Peaks/SPP/IDR/Pseudo-replicates'><td> Replicate 2 </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/IDR/Pseudo-replicates/Replicate 2/IDR peak' data-tt-parent-id='Peaks/SPP/IDR/Pseudo-replicates/Replicate 2'><td> IDR peak </td><td><a href='./peak/spp/idr/pseudo_reps/rep2/MITF_ANCG_Output_rep2-pr.IDR0.05.filt.narrowPeak.gz' target='_blank'>./peak/spp/idr/pseudo_reps/rep2/MITF_ANCG_Output_rep2-pr.IDR0.05.filt.narrowPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/IDR/Pooled pseudo-replicates' data-tt-parent-id='Peaks/SPP/IDR'><td> Pooled pseudo-replicates </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/IDR/Pooled pseudo-replicates/IDR peak' data-tt-parent-id='Peaks/SPP/IDR/Pooled pseudo-replicates'><td> IDR peak </td><td><a href='./peak/spp/idr/pooled_pseudo_reps/MITF_ANCG_Output_ppr.IDR0.05.filt.narrowPeak.gz' target='_blank'>./peak/spp/idr/pooled_pseudo_reps/MITF_ANCG_Output_ppr.IDR0.05.filt.narrowPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/IDR/Optimal set' data-tt-parent-id='Peaks/SPP/IDR'><td> Optimal set </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/IDR/Optimal set/IDR peak' data-tt-parent-id='Peaks/SPP/IDR/Optimal set'><td> IDR peak </td><td><a href='./peak/spp/idr/optimal_set/MITF_ANCG_Output_ppr.IDR0.05.filt.narrowPeak.gz' target='_blank'>./peak/spp/idr/optimal_set/MITF_ANCG_Output_ppr.IDR0.05.filt.narrowPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/SPP/IDR/Conservative set' data-tt-parent-id='Peaks/SPP/IDR'><td> Conservative set </td><td><br></td></tr><tr data-tt-id='Peaks/SPP/IDR/Conservative set/IDR peak' data-tt-parent-id='Peaks/SPP/IDR/Conservative set'><td> IDR peak </td><td><a href='./peak/spp/idr/conservative_set/MITF_ANCG_Output_rep1-rep2.IDR0.05.filt.narrowPeak.gz' target='_blank'>./peak/spp/idr/conservative_set/MITF_ANCG_Output_rep1-rep2.IDR0.05.filt.narrowPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/MACS2' data-tt-parent-id='Peaks'><td> MACS2 </td><td><br></td></tr><tr data-tt-id='Peaks/MACS2/Replicate 1' data-tt-parent-id='Peaks/MACS2'><td> Replicate 1 </td><td><br></td></tr><tr data-tt-id='Peaks/MACS2/Replicate 1/Narrow peak' data-tt-parent-id='Peaks/MACS2/Replicate 1'><td> Narrow peak </td><td><a href='./peak/macs2/rep1/30M.GSLv5-8_i7_160.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup.tagAlign.pval0.01.500K.narrowPeak.gz' target='_blank'>./peak/macs2/rep1/30M.GSLv5-8_i7_160.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup.tagAlign.pval0.01.500K.narrowPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/MACS2/Replicate 2' data-tt-parent-id='Peaks/MACS2'><td> Replicate 2 </td><td><br></td></tr><tr data-tt-id='Peaks/MACS2/Replicate 2/Narrow peak' data-tt-parent-id='Peaks/MACS2/Replicate 2'><td> Narrow peak </td><td><a href='./peak/macs2/rep2/30M.GSLv5-8_i7_165.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.pval0.01.500K.narrowPeak.gz' target='_blank'>./peak/macs2/rep2/30M.GSLv5-8_i7_165.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.pval0.01.500K.narrowPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/MACS2/Pooled replicate' data-tt-parent-id='Peaks/MACS2'><td> Pooled replicate </td><td><br></td></tr><tr data-tt-id='Peaks/MACS2/Pooled replicate/Narrow peak' data-tt-parent-id='Peaks/MACS2/Pooled replicate'><td> Narrow peak </td><td><a href='./peak/macs2/pooled_rep/30M.GSLv5-8_i7_160.merge.nodup_30M.GSLv5-8_i7_165.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.pval0.01.500K.narrowPeak.gz' target='_blank'>./peak/macs2/pooled_rep/30M.GSLv5-8_i7_160.merge.nodup_30M.GSLv5-8_i7_165.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.pval0.01.500K.narrowPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/MACS2/Pseudo-replicates' data-tt-parent-id='Peaks/MACS2'><td> Pseudo-replicates </td><td><br></td></tr><tr data-tt-id='Peaks/MACS2/Pseudo-replicates/Replicate 1' data-tt-parent-id='Peaks/MACS2/Pseudo-replicates'><td> Replicate 1 </td><td><br></td></tr><tr data-tt-id='Peaks/MACS2/Pseudo-replicates/Replicate 1/Pseudo-replicate 1' data-tt-parent-id='Peaks/MACS2/Pseudo-replicates/Replicate 1'><td> Pseudo-replicate 1 </td><td><br></td></tr><tr data-tt-id='Peaks/MACS2/Pseudo-replicates/Replicate 1/Pseudo-replicate 1/Narrow peak' data-tt-parent-id='Peaks/MACS2/Pseudo-replicates/Replicate 1/Pseudo-replicate 1'><td> Narrow peak </td><td><a href='./peak/macs2/pseudo_reps/rep1/pr1/30M.GSLv5-8_i7_160.merge.nodup.pr1.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup.tagAlign.pval0.01.500K.narrowPeak.gz' target='_blank'>./peak/macs2/pseudo_reps/rep1/pr1/30M.GSLv5-8_i7_160.merge.nodup.pr1.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup.tagAlign.pval0.01.500K.narrowPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/MACS2/Pseudo-replicates/Replicate 1/Pseudo-replicate 2' data-tt-parent-id='Peaks/MACS2/Pseudo-replicates/Replicate 1'><td> Pseudo-replicate 2 </td><td><br></td></tr><tr data-tt-id='Peaks/MACS2/Pseudo-replicates/Replicate 1/Pseudo-replicate 2/Narrow peak' data-tt-parent-id='Peaks/MACS2/Pseudo-replicates/Replicate 1/Pseudo-replicate 2'><td> Narrow peak </td><td><a href='./peak/macs2/pseudo_reps/rep1/pr2/30M.GSLv5-8_i7_160.merge.nodup.pr2.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup.tagAlign.pval0.01.500K.narrowPeak.gz' target='_blank'>./peak/macs2/pseudo_reps/rep1/pr2/30M.GSLv5-8_i7_160.merge.nodup.pr2.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup.tagAlign.pval0.01.500K.narrowPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/MACS2/Pseudo-replicates/Replicate 2' data-tt-parent-id='Peaks/MACS2/Pseudo-replicates'><td> Replicate 2 </td><td><br></td></tr><tr data-tt-id='Peaks/MACS2/Pseudo-replicates/Replicate 2/Pseudo-replicate 1' data-tt-parent-id='Peaks/MACS2/Pseudo-replicates/Replicate 2'><td> Pseudo-replicate 1 </td><td><br></td></tr><tr data-tt-id='Peaks/MACS2/Pseudo-replicates/Replicate 2/Pseudo-replicate 1/Narrow peak' data-tt-parent-id='Peaks/MACS2/Pseudo-replicates/Replicate 2/Pseudo-replicate 1'><td> Narrow peak </td><td><a href='./peak/macs2/pseudo_reps/rep2/pr1/30M.GSLv5-8_i7_165.merge.nodup.pr1.tagAlign_x_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.pval0.01.500K.narrowPeak.gz' target='_blank'>./peak/macs2/pseudo_reps/rep2/pr1/30M.GSLv5-8_i7_165.merge.nodup.pr1.tagAlign_x_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.pval0.01.500K.narrowPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/MACS2/Pseudo-replicates/Replicate 2/Pseudo-replicate 2' data-tt-parent-id='Peaks/MACS2/Pseudo-replicates/Replicate 2'><td> Pseudo-replicate 2 </td><td><br></td></tr><tr data-tt-id='Peaks/MACS2/Pseudo-replicates/Replicate 2/Pseudo-replicate 2/Narrow peak' data-tt-parent-id='Peaks/MACS2/Pseudo-replicates/Replicate 2/Pseudo-replicate 2'><td> Narrow peak </td><td><a href='./peak/macs2/pseudo_reps/rep2/pr2/30M.GSLv5-8_i7_165.merge.nodup.pr2.tagAlign_x_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.pval0.01.500K.narrowPeak.gz' target='_blank'>./peak/macs2/pseudo_reps/rep2/pr2/30M.GSLv5-8_i7_165.merge.nodup.pr2.tagAlign_x_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.pval0.01.500K.narrowPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/MACS2/Pooled pseudo-replicate' data-tt-parent-id='Peaks/MACS2'><td> Pooled pseudo-replicate </td><td><br></td></tr><tr data-tt-id='Peaks/MACS2/Pooled pseudo-replicate/Pooled pseudo-replicate 1' data-tt-parent-id='Peaks/MACS2/Pooled pseudo-replicate'><td> Pooled pseudo-replicate 1 </td><td><br></td></tr><tr data-tt-id='Peaks/MACS2/Pooled pseudo-replicate/Pooled pseudo-replicate 1/Narrow peak' data-tt-parent-id='Peaks/MACS2/Pooled pseudo-replicate/Pooled pseudo-replicate 1'><td> Narrow peak </td><td><a href='./peak/macs2/pooled_pseudo_reps/ppr1/30M.GSLv5-8_i7_160.merge.nodup.pr1_30M.GSLv5-8_i7_165.merge.nodup.pr1.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.pval0.01.500K.narrowPeak.gz' target='_blank'>./peak/macs2/pooled_pseudo_reps/ppr1/30M.GSLv5-8_i7_160.merge.nodup.pr1_30M.GSLv5-8_i7_165.merge.nodup.pr1.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.pval0.01.500K.narrowPeak.gz</a><br></td></tr><tr data-tt-id='Peaks/MACS2/Pooled pseudo-replicate/Pooled pseudo-replicate 2' data-tt-parent-id='Peaks/MACS2/Pooled pseudo-replicate'><td> Pooled pseudo-replicate 2 </td><td><br></td></tr><tr data-tt-id='Peaks/MACS2/Pooled pseudo-replicate/Pooled pseudo-replicate 2/Narrow peak' data-tt-parent-id='Peaks/MACS2/Pooled pseudo-replicate/Pooled pseudo-replicate 2'><td> Narrow peak </td><td><a href='./peak/macs2/pooled_pseudo_reps/ppr2/30M.GSLv5-8_i7_160.merge.nodup.pr2_30M.GSLv5-8_i7_165.merge.nodup.pr2.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.pval0.01.500K.narrowPeak.gz' target='_blank'>./peak/macs2/pooled_pseudo_reps/ppr2/30M.GSLv5-8_i7_160.merge.nodup.pr2_30M.GSLv5-8_i7_165.merge.nodup.pr2.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.pval0.01.500K.narrowPeak.gz</a><br></td></tr><tr data-tt-id='QC and logs'><td> QC and logs </td><td><br></td></tr><tr data-tt-id='QC and logs/Fingerprint JSD Plot' data-tt-parent-id='QC and logs'><td> Fingerprint JSD Plot </td><td><a href='./qc/MITF_ANCG_Output_jsd.png' target='_blank'>./qc/MITF_ANCG_Output_jsd.png</a><br></td></tr><tr data-tt-id='QC and logs/Replicate 1' data-tt-parent-id='QC and logs'><td> Replicate 1 </td><td><br></td></tr><tr data-tt-id='QC and logs/Replicate 1/Filtered flagstat log' data-tt-parent-id='QC and logs/Replicate 1'><td> Filtered flagstat log </td><td><a href='./qc/rep1/30M.GSLv5-8_i7_160.merge.nodup.flagstat.qc' target='_blank'>./qc/rep1/30M.GSLv5-8_i7_160.merge.nodup.flagstat.qc</a><br></td></tr><tr data-tt-id='QC and logs/Replicate 1/JS Distance' data-tt-parent-id='QC and logs/Replicate 1'><td> JS Distance </td><td><a href='./qc/rep1/30M.GSLv5-8_i7_160.merge.nodup.jsd.qc' target='_blank'>./qc/rep1/30M.GSLv5-8_i7_160.merge.nodup.jsd.qc</a><br></td></tr><tr data-tt-id='QC and logs/Replicate 1/BWA map. flagstat log' data-tt-parent-id='QC and logs/Replicate 1'><td> BWA map. flagstat log </td><td><a href='./qc/rep1/30M.GSLv5-8_i7_160.merge.flagstat.qc' target='_blank'>./qc/rep1/30M.GSLv5-8_i7_160.merge.flagstat.qc</a><br></td></tr><tr data-tt-id='QC and logs/Replicate 1/Dedup. log' data-tt-parent-id='QC and logs/Replicate 1'><td> Dedup. log </td><td><a href='./qc/rep1/30M.GSLv5-8_i7_160.merge.dup.qc' target='_blank'>./qc/rep1/30M.GSLv5-8_i7_160.merge.dup.qc</a><br></td></tr><tr data-tt-id='QC and logs/Replicate 1/PBC log' data-tt-parent-id='QC and logs/Replicate 1'><td> PBC log </td><td><a href='./qc/rep1/30M.GSLv5-8_i7_160.merge.nodup.pbc.qc' target='_blank'>./qc/rep1/30M.GSLv5-8_i7_160.merge.nodup.pbc.qc</a><br></td></tr><tr data-tt-id='QC and logs/Replicate 1/Cross-corr. log' data-tt-parent-id='QC and logs/Replicate 1'><td> Cross-corr. log </td><td><a href='./qc/rep1/30M.GSLv5-8_i7_160.merge.no_chrM.15M.cc.qc' target='_blank'>./qc/rep1/30M.GSLv5-8_i7_160.merge.no_chrM.15M.cc.qc</a><br></td></tr><tr data-tt-id='QC and logs/Replicate 1/Cross-corr. plot' data-tt-parent-id='QC and logs/Replicate 1'><td> Cross-corr. plot </td><td><a href='./qc/rep1/30M.GSLv5-8_i7_160.merge.no_chrM.15M.cc.plot.pdf' target='_blank'>./qc/rep1/30M.GSLv5-8_i7_160.merge.no_chrM.15M.cc.plot.pdf</a><br></td></tr><tr data-tt-id='QC and logs/Replicate 2' data-tt-parent-id='QC and logs'><td> Replicate 2 </td><td><br></td></tr><tr data-tt-id='QC and logs/Replicate 2/Filtered flagstat log' data-tt-parent-id='QC and logs/Replicate 2'><td> Filtered flagstat log </td><td><a href='./qc/rep2/30M.GSLv5-8_i7_165.merge.nodup.flagstat.qc' target='_blank'>./qc/rep2/30M.GSLv5-8_i7_165.merge.nodup.flagstat.qc</a><br></td></tr><tr data-tt-id='QC and logs/Replicate 2/JS Distance' data-tt-parent-id='QC and logs/Replicate 2'><td> JS Distance </td><td><a href='./qc/rep2/30M.GSLv5-8_i7_165.merge.nodup.jsd.qc' target='_blank'>./qc/rep2/30M.GSLv5-8_i7_165.merge.nodup.jsd.qc</a><br></td></tr><tr data-tt-id='QC and logs/Replicate 2/BWA map. flagstat log' data-tt-parent-id='QC and logs/Replicate 2'><td> BWA map. flagstat log </td><td><a href='./qc/rep2/30M.GSLv5-8_i7_165.merge.flagstat.qc' target='_blank'>./qc/rep2/30M.GSLv5-8_i7_165.merge.flagstat.qc</a><br></td></tr><tr data-tt-id='QC and logs/Replicate 2/Dedup. log' data-tt-parent-id='QC and logs/Replicate 2'><td> Dedup. log </td><td><a href='./qc/rep2/30M.GSLv5-8_i7_165.merge.dup.qc' target='_blank'>./qc/rep2/30M.GSLv5-8_i7_165.merge.dup.qc</a><br></td></tr><tr data-tt-id='QC and logs/Replicate 2/PBC log' data-tt-parent-id='QC and logs/Replicate 2'><td> PBC log </td><td><a href='./qc/rep2/30M.GSLv5-8_i7_165.merge.nodup.pbc.qc' target='_blank'>./qc/rep2/30M.GSLv5-8_i7_165.merge.nodup.pbc.qc</a><br></td></tr><tr data-tt-id='QC and logs/Replicate 2/Cross-corr. log' data-tt-parent-id='QC and logs/Replicate 2'><td> Cross-corr. log </td><td><a href='./qc/rep2/30M.GSLv5-8_i7_165.merge.no_chrM.15M.cc.qc' target='_blank'>./qc/rep2/30M.GSLv5-8_i7_165.merge.no_chrM.15M.cc.qc</a><br></td></tr><tr data-tt-id='QC and logs/Replicate 2/Cross-corr. plot' data-tt-parent-id='QC and logs/Replicate 2'><td> Cross-corr. plot </td><td><a href='./qc/rep2/30M.GSLv5-8_i7_165.merge.no_chrM.15M.cc.plot.pdf' target='_blank'>./qc/rep2/30M.GSLv5-8_i7_165.merge.no_chrM.15M.cc.plot.pdf</a><br></td></tr><tr data-tt-id='QC and logs/Control 1' data-tt-parent-id='QC and logs'><td> Control 1 </td><td><br></td></tr><tr data-tt-id='QC and logs/Control 1/Filtered flagstat log' data-tt-parent-id='QC and logs/Control 1'><td> Filtered flagstat log </td><td><a href='./qc/ctl1/30M.GSLv5-8_i7_170.merge.nodup.flagstat.qc' target='_blank'>./qc/ctl1/30M.GSLv5-8_i7_170.merge.nodup.flagstat.qc</a><br></td></tr><tr data-tt-id='QC and logs/Control 1/BWA map. flagstat log' data-tt-parent-id='QC and logs/Control 1'><td> BWA map. flagstat log </td><td><a href='./qc/ctl1/30M.GSLv5-8_i7_170.merge.flagstat.qc' target='_blank'>./qc/ctl1/30M.GSLv5-8_i7_170.merge.flagstat.qc</a><br></td></tr><tr data-tt-id='QC and logs/Control 1/Dedup. log' data-tt-parent-id='QC and logs/Control 1'><td> Dedup. log </td><td><a href='./qc/ctl1/30M.GSLv5-8_i7_170.merge.dup.qc' target='_blank'>./qc/ctl1/30M.GSLv5-8_i7_170.merge.dup.qc</a><br></td></tr><tr data-tt-id='QC and logs/Control 1/PBC log' data-tt-parent-id='QC and logs/Control 1'><td> PBC log </td><td><a href='./qc/ctl1/30M.GSLv5-8_i7_170.merge.nodup.pbc.qc' target='_blank'>./qc/ctl1/30M.GSLv5-8_i7_170.merge.nodup.pbc.qc</a><br></td></tr><tr data-tt-id='QC and logs/Control 2' data-tt-parent-id='QC and logs'><td> Control 2 </td><td><br></td></tr><tr data-tt-id='QC and logs/Control 2/Filtered flagstat log' data-tt-parent-id='QC and logs/Control 2'><td> Filtered flagstat log </td><td><a href='./qc/ctl2/30M.GSLv5-8_i7_175.merge.nodup.flagstat.qc' target='_blank'>./qc/ctl2/30M.GSLv5-8_i7_175.merge.nodup.flagstat.qc</a><br></td></tr><tr data-tt-id='QC and logs/Control 2/BWA map. flagstat log' data-tt-parent-id='QC and logs/Control 2'><td> BWA map. flagstat log </td><td><a href='./qc/ctl2/30M.GSLv5-8_i7_175.merge.flagstat.qc' target='_blank'>./qc/ctl2/30M.GSLv5-8_i7_175.merge.flagstat.qc</a><br></td></tr><tr data-tt-id='QC and logs/Control 2/Dedup. log' data-tt-parent-id='QC and logs/Control 2'><td> Dedup. log </td><td><a href='./qc/ctl2/30M.GSLv5-8_i7_175.merge.dup.qc' target='_blank'>./qc/ctl2/30M.GSLv5-8_i7_175.merge.dup.qc</a><br></td></tr><tr data-tt-id='QC and logs/Control 2/PBC log' data-tt-parent-id='QC and logs/Control 2'><td> PBC log </td><td><a href='./qc/ctl2/30M.GSLv5-8_i7_175.merge.nodup.pbc.qc' target='_blank'>./qc/ctl2/30M.GSLv5-8_i7_175.merge.nodup.pbc.qc</a><br></td></tr><tr data-tt-id='QC and logs/Naive overlap' data-tt-parent-id='QC and logs'><td> Naive overlap </td><td><br></td></tr><tr data-tt-id='QC and logs/Naive overlap/Overlap QC log' data-tt-parent-id='QC and logs/Naive overlap'><td> Overlap QC log </td><td><a href='./qc/MITF_ANCG_Output_peak_overlap_final.qc' target='_blank'>./qc/MITF_ANCG_Output_peak_overlap_final.qc</a><br></td></tr><tr data-tt-id='QC and logs/Naive overlap/True replicates' data-tt-parent-id='QC and logs/Naive overlap'><td> True replicates </td><td><br></td></tr><tr data-tt-id='QC and logs/Naive overlap/True replicates/Rep. 1 vs. Rep. 2' data-tt-parent-id='QC and logs/Naive overlap/True replicates'><td> Rep. 1 vs. Rep. 2 </td><td><br></td></tr><tr data-tt-id='QC and logs/Naive overlap/True replicates/Rep. 1 vs. Rep. 2/FRiP' data-tt-parent-id='QC and logs/Naive overlap/True replicates/Rep. 1 vs. Rep. 2'><td> FRiP </td><td><a href='./peak/spp/overlap/true_reps/rep1-rep2/MITF_ANCG_Output_rep1-rep2.naive_overlap.filt.regionPeak.FRiP.qc' target='_blank'>./peak/spp/overlap/true_reps/rep1-rep2/MITF_ANCG_Output_rep1-rep2.naive_overlap.filt.regionPeak.FRiP.qc</a><br></td></tr><tr data-tt-id='QC and logs/Naive overlap/Pseudo-replicates' data-tt-parent-id='QC and logs/Naive overlap'><td> Pseudo-replicates </td><td><br></td></tr><tr data-tt-id='QC and logs/Naive overlap/Pseudo-replicates/Replicate 1' data-tt-parent-id='QC and logs/Naive overlap/Pseudo-replicates'><td> Replicate 1 </td><td><br></td></tr><tr data-tt-id='QC and logs/Naive overlap/Pseudo-replicates/Replicate 1/FRiP' data-tt-parent-id='QC and logs/Naive overlap/Pseudo-replicates/Replicate 1'><td> FRiP </td><td><a href='./peak/spp/overlap/pseudo_reps/rep1/MITF_ANCG_Output_rep1-pr.naive_overlap.filt.regionPeak.FRiP.qc' target='_blank'>./peak/spp/overlap/pseudo_reps/rep1/MITF_ANCG_Output_rep1-pr.naive_overlap.filt.regionPeak.FRiP.qc</a><br></td></tr><tr data-tt-id='QC and logs/Naive overlap/Pseudo-replicates/Replicate 2' data-tt-parent-id='QC and logs/Naive overlap/Pseudo-replicates'><td> Replicate 2 </td><td><br></td></tr><tr data-tt-id='QC and logs/Naive overlap/Pseudo-replicates/Replicate 2/FRiP' data-tt-parent-id='QC and logs/Naive overlap/Pseudo-replicates/Replicate 2'><td> FRiP </td><td><a href='./peak/spp/overlap/pseudo_reps/rep2/MITF_ANCG_Output_rep2-pr.naive_overlap.filt.regionPeak.FRiP.qc' target='_blank'>./peak/spp/overlap/pseudo_reps/rep2/MITF_ANCG_Output_rep2-pr.naive_overlap.filt.regionPeak.FRiP.qc</a><br></td></tr><tr data-tt-id='QC and logs/IDR' data-tt-parent-id='QC and logs'><td> IDR </td><td><br></td></tr><tr data-tt-id='QC and logs/IDR/IDR QC log' data-tt-parent-id='QC and logs/IDR'><td> IDR QC log </td><td><a href='./qc/MITF_ANCG_Output_IDR_final.qc' target='_blank'>./qc/MITF_ANCG_Output_IDR_final.qc</a><br></td></tr><tr data-tt-id='QC and logs/IDR/True replicates' data-tt-parent-id='QC and logs/IDR'><td> True replicates </td><td><br></td></tr><tr data-tt-id='QC and logs/IDR/True replicates/Rep. 1 vs. Rep. 2' data-tt-parent-id='QC and logs/IDR/True replicates'><td> Rep. 1 vs. Rep. 2 </td><td><br></td></tr><tr data-tt-id='QC and logs/IDR/True replicates/Rep. 1 vs. Rep. 2/FRiP' data-tt-parent-id='QC and logs/IDR/True replicates/Rep. 1 vs. Rep. 2'><td> FRiP </td><td><a href='./peak/spp/idr/true_reps/rep1-rep2/MITF_ANCG_Output_rep1-rep2.IDR0.05.filt.narrowPeak.FRiP.qc' target='_blank'>./peak/spp/idr/true_reps/rep1-rep2/MITF_ANCG_Output_rep1-rep2.IDR0.05.filt.narrowPeak.FRiP.qc</a><br></td></tr><tr data-tt-id='QC and logs/IDR/True replicates/Rep. 1 vs. Rep. 2/IDR plot' data-tt-parent-id='QC and logs/IDR/True replicates/Rep. 1 vs. Rep. 2'><td> IDR plot </td><td><a href='./peak/spp/idr/true_reps/rep1-rep2/MITF_ANCG_Output_rep1-rep2.unthresholded-peaks.txt.png' target='_blank'>./peak/spp/idr/true_reps/rep1-rep2/MITF_ANCG_Output_rep1-rep2.unthresholded-peaks.txt.png</a><br></td></tr><tr data-tt-id='QC and logs/IDR/Pseudo-replicates' data-tt-parent-id='QC and logs/IDR'><td> Pseudo-replicates </td><td><br></td></tr><tr data-tt-id='QC and logs/IDR/Pseudo-replicates/Replicate 1' data-tt-parent-id='QC and logs/IDR/Pseudo-replicates'><td> Replicate 1 </td><td><br></td></tr><tr data-tt-id='QC and logs/IDR/Pseudo-replicates/Replicate 1/FRiP' data-tt-parent-id='QC and logs/IDR/Pseudo-replicates/Replicate 1'><td> FRiP </td><td><a href='./peak/spp/idr/pseudo_reps/rep1/MITF_ANCG_Output_rep1-pr.IDR0.05.filt.narrowPeak.FRiP.qc' target='_blank'>./peak/spp/idr/pseudo_reps/rep1/MITF_ANCG_Output_rep1-pr.IDR0.05.filt.narrowPeak.FRiP.qc</a><br></td></tr><tr data-tt-id='QC and logs/IDR/Pseudo-replicates/Replicate 1/IDR plot' data-tt-parent-id='QC and logs/IDR/Pseudo-replicates/Replicate 1'><td> IDR plot </td><td><a href='./peak/spp/idr/pseudo_reps/rep1/MITF_ANCG_Output_rep1-pr.unthresholded-peaks.txt.png' target='_blank'>./peak/spp/idr/pseudo_reps/rep1/MITF_ANCG_Output_rep1-pr.unthresholded-peaks.txt.png</a><br></td></tr><tr data-tt-id='QC and logs/IDR/Pseudo-replicates/Replicate 2' data-tt-parent-id='QC and logs/IDR/Pseudo-replicates'><td> Replicate 2 </td><td><br></td></tr><tr data-tt-id='QC and logs/IDR/Pseudo-replicates/Replicate 2/FRiP' data-tt-parent-id='QC and logs/IDR/Pseudo-replicates/Replicate 2'><td> FRiP </td><td><a href='./peak/spp/idr/pseudo_reps/rep2/MITF_ANCG_Output_rep2-pr.IDR0.05.filt.narrowPeak.FRiP.qc' target='_blank'>./peak/spp/idr/pseudo_reps/rep2/MITF_ANCG_Output_rep2-pr.IDR0.05.filt.narrowPeak.FRiP.qc</a><br></td></tr><tr data-tt-id='QC and logs/IDR/Pseudo-replicates/Replicate 2/IDR plot' data-tt-parent-id='QC and logs/IDR/Pseudo-replicates/Replicate 2'><td> IDR plot </td><td><a href='./peak/spp/idr/pseudo_reps/rep2/MITF_ANCG_Output_rep2-pr.unthresholded-peaks.txt.png' target='_blank'>./peak/spp/idr/pseudo_reps/rep2/MITF_ANCG_Output_rep2-pr.unthresholded-peaks.txt.png</a><br></td></tr><tr data-tt-id='QC and logs/IDR/Pooled pseudo-replicates' data-tt-parent-id='QC and logs/IDR'><td> Pooled pseudo-replicates </td><td><br></td></tr><tr data-tt-id='QC and logs/IDR/Pooled pseudo-replicates/IDR plot' data-tt-parent-id='QC and logs/IDR/Pooled pseudo-replicates'><td> IDR plot </td><td><a href='./peak/spp/idr/pooled_pseudo_reps/MITF_ANCG_Output_ppr.unthresholded-peaks.txt.png' target='_blank'>./peak/spp/idr/pooled_pseudo_reps/MITF_ANCG_Output_ppr.unthresholded-peaks.txt.png</a><br></td></tr></tbody></table></div><br>
<div id='viz'><b>Visualization (add <i>-url_base [URL_ROOT_DIR_FOR_OUT_DIR]</i> to the command line.) </b><div style='padding-left: 20px'><a href=http://epigenomegateway.wustl.edu/browser/?genome=hg38&tknamewidth=275&datahub=/./MITF_ANCG_Output_tracks.json target='_blank'>Visualize</a>&nbsp&nbsp<a href=./MITF_ANCG_Output_tracks.json target='_blank'>JSON (datahub)</a></div></div><br>

<button id='btn_toggle'>Toggle workflow</button><br><br>
<script>$('#btn_toggle').click(function() {$('#graph').toggle();});</script><div id='graph'><b>Workflow diagram</b>&nbsp<small>(g./b. peak : gapped/broad peak from MACS2, n. peak : narrow peak from MACS2, r. peak : relaxed peak from SPP)</small><br><object data='./report/MITF_ANCG_Output_workflow.svg'></object></div><br></div><br>
<div id='chipseq_qc'><div style='float:left'><div id='table_Flagstat-(raw)-QC'><b>Flagstat (raw) QC</b>
<table border='1' style='margin-right:15px;margin-bottom:15px'>
<tr><th nowrap align=left>&nbsp</th><th nowrap align=left><a href='./qc/rep1/30M.GSLv5-8_i7_160.merge.flagstat.qc' target='_blank'>rep1</a><br></th><th nowrap align=left><a href='./qc/rep2/30M.GSLv5-8_i7_165.merge.flagstat.qc' target='_blank'>rep2</a><br></th><th nowrap align=left><a href='./qc/ctl1/30M.GSLv5-8_i7_170.merge.flagstat.qc' target='_blank'>ctl1</a><br></th><th nowrap align=left><a href='./qc/ctl2/30M.GSLv5-8_i7_175.merge.flagstat.qc' target='_blank'>ctl2</a><br></th></tr>
<tr><th nowrap align=left>Total</th><td>29308344</td><td>30000000</td><td>26638334</td><td>30000000</td></tr>
<tr><th nowrap align=left>Total(QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Dupes</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Dupes(QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Mapped</th><td>24447043</td><td>25761870</td><td>25987524</td><td>29232942</td></tr>
<tr><th nowrap align=left>Mapped(QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>% Mapped</th><td>83.41</td><td>85.87</td><td>97.56</td><td>97.44</td></tr>
<tr><th nowrap align=left>Paired</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Paired(QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Read1</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Read1(QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Read2</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Read2(QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Properly Paired</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Properly Paired(QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>% Properly Paired</th><td>-nan</td><td>-nan</td><td>-nan</td><td>-nan</td></tr>
<tr><th nowrap align=left>With itself</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>With itself(QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Singletons</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Singletons(QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>% Singleton</th><td>-nan</td><td>-nan</td><td>-nan</td><td>-nan</td></tr>
<tr><th nowrap align=left>Diff. Chroms</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Diff. Chroms (QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
</table></div></div><div style='float:left'><div id='table_Flagstat-(filtered)-QC'><b>Flagstat (filtered) QC</b>
<table border='1' style='margin-right:15px;margin-bottom:15px'>
<tr><th nowrap align=left>&nbsp</th><th nowrap align=left><a href='./qc/rep1/30M.GSLv5-8_i7_160.merge.nodup.flagstat.qc' target='_blank'>rep1</a><br></th><th nowrap align=left><a href='./qc/rep2/30M.GSLv5-8_i7_165.merge.nodup.flagstat.qc' target='_blank'>rep2</a><br></th><th nowrap align=left><a href='./qc/ctl1/30M.GSLv5-8_i7_170.merge.nodup.flagstat.qc' target='_blank'>ctl1</a><br></th><th nowrap align=left><a href='./qc/ctl2/30M.GSLv5-8_i7_175.merge.nodup.flagstat.qc' target='_blank'>ctl2</a><br></th></tr>
<tr><th nowrap align=left>Total</th><td>16600160</td><td>18994595</td><td>21742592</td><td>24572355</td></tr>
<tr><th nowrap align=left>Total(QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Dupes</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Dupes(QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Mapped</th><td>16600160</td><td>18994595</td><td>21742592</td><td>24572355</td></tr>
<tr><th nowrap align=left>Mapped(QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>% Mapped</th><td>100.00</td><td>100.00</td><td>100.00</td><td>100.00</td></tr>
<tr><th nowrap align=left>Paired</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Paired(QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Read1</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Read1(QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Read2</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Read2(QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Properly Paired</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Properly Paired(QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>% Properly Paired</th><td>N/A</td><td>N/A</td><td>N/A</td><td>N/A</td></tr>
<tr><th nowrap align=left>With itself</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>With itself(QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Singletons</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Singletons(QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>% Singleton</th><td>N/A</td><td>N/A</td><td>N/A</td><td>N/A</td></tr>
<tr><th nowrap align=left>Diff. Chroms</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Diff. Chroms (QC-failed)</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
</table></div></div><div style='float:left'><div id='table_Dup.-QC'><b>Dup. QC</b>
<table border='1' style='margin-right:15px;margin-bottom:15px'>
<tr><th nowrap align=left>&nbsp</th><th nowrap align=left><a href='./qc/rep1/30M.GSLv5-8_i7_160.merge.dup.qc' target='_blank'>rep1</a><br></th><th nowrap align=left><a href='./qc/rep2/30M.GSLv5-8_i7_165.merge.dup.qc' target='_blank'>rep2</a><br></th><th nowrap align=left><a href='./qc/ctl1/30M.GSLv5-8_i7_170.merge.dup.qc' target='_blank'>ctl1</a><br></th><th nowrap align=left><a href='./qc/ctl2/30M.GSLv5-8_i7_175.merge.dup.qc' target='_blank'>ctl2</a><br></th></tr>
<tr><th nowrap align=left>Unpaired Reads</th><td>22097673</td><td>23161293</td><td>22621325</td><td>25502144</td></tr>
<tr><th nowrap align=left>Paired Reads</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Unmapped Reads</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Unpaired Dupes</th><td>5497513</td><td>4166698</td><td>878733</td><td>929789</td></tr>
<tr><th nowrap align=left>Paired Dupes</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>Paired Opt. Dupes</th><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th nowrap align=left>% Dupes/100</th><td>0.248782</td><td>0.179899</td><td>0.038845</td><td>0.036459</td></tr>
</table></div></div><div style='float:left'><div id='table_Fingerprint-JS-metric'><b>Fingerprint JS metric</b>
<table border='1' style='margin-right:15px;margin-bottom:15px'>
<tr><th nowrap align=left>&nbsp</th><th nowrap align=left><a href='./qc/rep1/30M.GSLv5-8_i7_160.merge.nodup.jsd.qc' target='_blank'>rep1</a><br></th><th nowrap align=left><a href='./qc/rep2/30M.GSLv5-8_i7_165.merge.nodup.jsd.qc' target='_blank'>rep2</a><br></th></tr>
<tr><th nowrap align=left>% genome enriched</th><td>0.273796376353</td><td>0.291574739325</td></tr>
<tr><th nowrap align=left>AUC</th><td>0.49117835295</td><td>0.491748536901</td></tr>
<tr><th nowrap align=left>CHANCE divergence</th><td>0.155201989708</td><td>0.131896100157</td></tr>
<tr><th nowrap align=left>Elbow Point</th><td>1.982277134e-109</td><td>2.92426714672e-125</td></tr>
<tr><th nowrap align=left>JS Distance</th><td>0.556194444333</td><td>0.536205400521</td></tr>
<tr><th nowrap align=left>Synthetic AUC</th><td>0.495528049718</td><td>0.50548186673</td></tr>
<tr><th nowrap align=left>Synthetic Elbow Point</th><td>0.0570719174444</td><td>0.0262012573398</td></tr>
<tr><th nowrap align=left>Synthetic JS Distance</th><td>0.251480167653</td><td>0.233696848318</td></tr>
<tr><th nowrap align=left>Synthetic X-intercept</th><td>38.9568430186</td><td>37.7736547681</td></tr>
<tr><th nowrap align=left>X-intercept</th><td>24.8787383356</td><td>22.286291894</td></tr>
<tr><th nowrap align=left>diff. enrichment</th><td>0.215764097063</td><td>0.191438010976</td></tr>
</table></div><figure style='display: inline-block;'><img src='./qc/MITF_ANCG_Output_jsd.png' width='800'><figcaption style='text-align: center;'><b>Fingerprint plot</b></figcaption></figure><br></div><div style='float:left'><div id='table_Library-Complexity-QC'><b>Library Complexity QC</b>
<table border='1' style='margin-right:15px;margin-bottom:15px'>
<tr><th nowrap align=left>&nbsp</th><th nowrap align=left><a href='./qc/rep1/30M.GSLv5-8_i7_160.merge.nodup.pbc.qc' target='_blank'>rep1</a><br></th><th nowrap align=left><a href='./qc/rep2/30M.GSLv5-8_i7_165.merge.nodup.pbc.qc' target='_blank'>rep2</a><br></th><th nowrap align=left><a href='./qc/ctl1/30M.GSLv5-8_i7_170.merge.nodup.pbc.qc' target='_blank'>ctl1</a><br></th><th nowrap align=left><a href='./qc/ctl2/30M.GSLv5-8_i7_175.merge.nodup.pbc.qc' target='_blank'>ctl2</a><br></th></tr>
<tr><th nowrap align=left>Total Reads</th><td>22018728</td><td>23084026</td><td>22474242</td><td>25276111</td></tr>
<tr><th nowrap align=left>Distinct Reads</th><td>16629632</td><td>19016946</td><td>21788769</td><td>24634322</td></tr>
<tr><th nowrap align=left>One Read</th><td>12365158</td><td>15559394</td><td>21126809</td><td>24018233</td></tr>
<tr><th nowrap align=left>Two Reads</th><td>3359194</td><td>2938904</td><td>645625</td><td>601572</td></tr>
<tr><th nowrap align=left>NRF = Distinct/Total</th><td>0.755249</td><td>0.823814</td><td>0.969500</td><td>0.974609</td></tr>
<tr><th nowrap align=left>PBC1 = OneRead/Distinct</th><td>0.743562</td><td>0.818186</td><td>0.969619</td><td>0.974991</td></tr>
<tr><th nowrap align=left>PBC2 = OneRead/TwoReads</th><td>3.680990</td><td>5.294285</td><td>32.723034</td><td>39.925783</td></tr>
</table></div><div id='help-pbc'><p>NRF (non redundant fraction) <br>PBC1 (PCR Bottleneck coefficient 1) <br>PBC2 (PCR Bottleneck coefficient 2) <br>PBC1 is the primary measure. Provisionally <br><ul><li>0-0.5 is severe bottlenecking</li><li>0.5-0.8 is moderate bottlenecking </li><li>0.8-0.9 is mild bottlenecking </li><li>0.9-1.0 is no bottlenecking </li></ul></p></div></div><div style='float:left'><div id='table_Enrichment-QC-(strand-cross-correlation-measures)'><b>Enrichment QC (strand cross-correlation measures)</b>
<table border='1' style='margin-right:15px;margin-bottom:15px'>
<tr><th nowrap align=left>&nbsp</th><th nowrap align=left><a href='./qc/rep1/30M.GSLv5-8_i7_160.merge.no_chrM.15M.cc.qc' target='_blank'>rep1</a><br></th><th nowrap align=left><a href='./qc/rep2/30M.GSLv5-8_i7_165.merge.no_chrM.15M.cc.qc' target='_blank'>rep2</a><br></th></tr>
<tr><th nowrap align=left>Reads</th><td>15000000</td><td>15000000</td></tr>
<tr><th nowrap align=left>Est. Fragment Len.</th><td>220</td><td>235</td></tr>
<tr><th nowrap align=left>Corr. Est. Fragment Len.</th><td>0.140088394102642</td><td>0.1491376226896</td></tr>
<tr><th nowrap align=left>Phantom Peak</th><td>100</td><td>100</td></tr>
<tr><th nowrap align=left>Corr. Phantom Peak</th><td>0.1518086</td><td>0.1624727</td></tr>
<tr><th nowrap align=left>Argmin. Corr.</th><td>1500</td><td>1500</td></tr>
<tr><th nowrap align=left>Min. Corr.</th><td>0.132351</td><td>0.1442497</td></tr>
<tr><th nowrap align=left>NSC</th><td>1.058461</td><td>1.033885</td></tr>
<tr><th nowrap align=left>RSC</th><td>0.3976541</td><td>0.2682265</td></tr>
</table></div><div id='help-xcor'><p>NOTE: Reads from replicates are subsampled to a max of 15M <br><ul><li>Normalized strand cross-correlation coefficient (NSC) = col9 in outFile </li><li>Relative strand cross-correlation coefficient (RSC) = col10 in outFile </li><li>Estimated fragment length = col3 in outFile, take the top value </li></ul></p><br></div><figure style='display: inline-block;'><img src='./qc/rep1/30M.GSLv5-8_i7_160.merge.no_chrM.15M.cc.plot.png' width='500'><figcaption style='text-align: center;'><b>rep1</b></figcaption></figure>&nbsp<figure style='display: inline-block;'><img src='./qc/rep2/30M.GSLv5-8_i7_165.merge.no_chrM.15M.cc.plot.png' width='500'><figcaption style='text-align: center;'><b>rep2</b></figcaption></figure>&nbsp<div style='float:left'><div id='table_Number-of-peaks'><b>Number of peaks</b>
<table border='1' style='margin-right:15px;margin-bottom:15px'>
<tr><th nowrap align=left>rep1</th><td>300000</td></tr>
<tr><th nowrap align=left>rep1-pr1</th><td>300000</td></tr>
<tr><th nowrap align=left>rep1-pr2</th><td>300000</td></tr>
<tr><th nowrap align=left>rep2</th><td>300000</td></tr>
<tr><th nowrap align=left>rep2-pr1</th><td>300000</td></tr>
<tr><th nowrap align=left>rep2-pr2</th><td>300000</td></tr>
<tr><th nowrap align=left>pooled</th><td>300000</td></tr>
<tr><th nowrap align=left>ppr1</th><td>300000</td></tr>
<tr><th nowrap align=left>ppr2</th><td>300000</td></tr>
</table></div><div id='help-num-peaks'><p><ul><li>ppr1: Raw peaks called on the first pooled pseudoreplicates </li><li>ppr2: Raw peaks called on the second pooled pseudoreplicates </li><li>repi: Raw peaks called on true replicate i </li><li>repi-pr1 : Raw peaks called on the first pseudoreplicate from replicate i </li><li>repi-pr2 : Raw peaks called on the second pseudoreplicates from replicate i </li><li>overlap : Overlapped peaks (filtered if blacklist exists) </li></ul></p><br></div><div style='float:left'><div id='table_Enrichment-QC-(Fraction-of-reads-in-raw-peaks)'><b>Enrichment QC (Fraction of reads in raw peaks)</b>
<table border='1' style='margin-right:15px;margin-bottom:15px'>
<tr><th nowrap align=left>&nbsp</th><th nowrap align=left><a href='./peak/spp/pooled_rep/30M.GSLv5-8_i7_160.merge.nodup_30M.GSLv5-8_i7_165.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.filt.regionPeak.FRiP.qc' target='_blank'>pooled_rep</a><br></th><th nowrap align=left><a href='./peak/spp/rep1/30M.GSLv5-8_i7_160.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_170.merge.nodup.tagAlign.filt.regionPeak.FRiP.qc' target='_blank'>rep1</a><br></th><th nowrap align=left><a href='./peak/spp/rep2/30M.GSLv5-8_i7_165.merge.nodup.tagAlign_x_30M.GSLv5-8_i7_175.merge.nodup.tagAlign.filt.regionPeak.FRiP.qc' target='_blank'>rep2</a><br></th></tr>
<tr><th nowrap align=left>Fraction of Reads in Peak</th><td>0.120371</td><td>0.149371</td><td>0.131243</td></tr>
</table></div></div><div style='float:left'><div id='table_Enrichment-QC-(Fraction-of-reads-in-overlapping-peaks)'><b>Enrichment QC (Fraction of reads in overlapping peaks)</b>
<table border='1' style='margin-right:15px;margin-bottom:15px'>
<tr><th nowrap align=left>&nbsp</th><th nowrap align=left><a href='./peak/spp/overlap/pseudo_reps/rep1/MITF_ANCG_Output_rep1-pr.naive_overlap.filt.regionPeak.FRiP.qc' target='_blank'>rep1-pr</a><br></th><th nowrap align=left><a href='./peak/spp/overlap/true_reps/rep1-rep2/MITF_ANCG_Output_rep1-rep2.naive_overlap.filt.regionPeak.FRiP.qc' target='_blank'>rep1-rep2</a><br></th><th nowrap align=left><a href='./peak/spp/overlap/pseudo_reps/rep2/MITF_ANCG_Output_rep2-pr.naive_overlap.filt.regionPeak.FRiP.qc' target='_blank'>rep2-pr</a><br></th></tr>
<tr><th nowrap align=left>Fraction of Reads in Peak</th><td>0.0369483</td><td>0.0264139</td><td>0.0231543</td></tr>
</table></div><div id='help-overlap-FRiP'><p><ul><li>ppr: Overlapping peaks comparing pooled pseudo replicates </li><li>rep1-pr: Overlapping peaks comparing pseudoreplicates from replicate 1 </li><li>rep2-pr: Overlapping peaks comparing pseudoreplicates from replicate 2 </li><li>repi-repj: Overlapping peaks comparing true replicates (rep i vs. rep j) </li></ul></p><br></div><div style='float:left'><div id='table_Reproducibility-QC-and-Peak-Detection-Statistics-(Overlapping-peaks)'><b>Reproducibility QC and Peak Detection Statistics (Overlapping peaks)</b>
<table border='1' style='margin-right:15px;margin-bottom:15px'>
<tr><th nowrap align=left>Nt</th><td>34691</td></tr>
<tr><th nowrap align=left>N1</th><td>39805</td></tr>
<tr><th nowrap align=left>N2</th><td>35284</td></tr>
<tr><th nowrap align=left>Np</th><td>47749</td></tr>
<tr><th nowrap align=left>N optimal</th><td>47749</td></tr>
<tr><th nowrap align=left>N conservative</th><td>34691</td></tr>
<tr><th nowrap align=left>Optimal Set</th><td>pooled_pseudo_rep</td></tr>
<tr><th nowrap align=left>Conservative Set</th><td>rep1-rep2</td></tr>
<tr><th nowrap align=left>Rescue Ratio</th><td>1.3764088668530743</td></tr>
<tr><th nowrap align=left>Self Consistency Ratio</th><td>1.1281317310962475</td></tr>
<tr><th nowrap align=left>Reproducibility Test</th><td>pass</td></tr>
</table></div><div id='help-overlap'><p><ul><li>N1: Replicate 1 self-consistent overlapping peaks (comparing two pseudoreplicates generated by subsampling Rep1 reads) </li><li>N2: Replicate 2 self-consistent overlapping peaks (comparing two pseudoreplicates generated by subsampling Rep2 reads) </li><li>Nt: True Replicate consisten overlapping peaks (comparing true replicates Rep1 vs Rep2 ) </li><li>Np: Pooled-pseudoreplicate consistent overlapping peaks (comparing two pseudoreplicates generated by subsampling pooled reads from Rep1 and Rep2 ) </li><li>Self-consistency Ratio: max(N1,N2) / min (N1,N2) </li><li>Rescue Ratio: max(Np,Nt) / min (Np,Nt) </li><li>Reproducibility Test: If Self-consistency Ratio >2 AND Rescue Ratio > 2, then 'Fail' else 'Pass' </li></ul></p><br></div><div style='float:left'><div id='table_Enrichment-QC-(Fraction-of-reads-in-IDR-peaks)'><b>Enrichment QC (Fraction of reads in IDR peaks)</b>
<table border='1' style='margin-right:15px;margin-bottom:15px'>
<tr><th nowrap align=left>&nbsp</th><th nowrap align=left><a href='./peak/spp/idr/pseudo_reps/rep1/MITF_ANCG_Output_rep1-pr.IDR0.05.filt.narrowPeak.FRiP.qc' target='_blank'>rep1-pr</a><br></th><th nowrap align=left><a href='./peak/spp/idr/true_reps/rep1-rep2/MITF_ANCG_Output_rep1-rep2.IDR0.05.filt.narrowPeak.FRiP.qc' target='_blank'>rep1-rep2</a><br></th><th nowrap align=left><a href='./peak/spp/idr/pseudo_reps/rep2/MITF_ANCG_Output_rep2-pr.IDR0.05.filt.narrowPeak.FRiP.qc' target='_blank'>rep2-pr</a><br></th></tr>
<tr><th nowrap align=left>Fraction of Reads in Peak</th><td>0.0158557</td><td>0.0110428</td><td>0.00439385</td></tr>
</table></div><div id='help-idr-FRiP'><p><ul><li>ppr: IDR peaks comparing pooled pseudo replicates </li><li>rep1-pr: IDR peaks comparing pseudoreplicates from replicate 1 </li><li>rep2-pr: IDR peaks comparing pseudoreplicates from replicate 2 </li><li>repi-repj: IDR peaks comparing true replicates (rep i vs. rep j) </li></ul></p><br></div><div style='float:left'><div id='table_Reproducibility-QC-and-Peak-Detection-Statistics-(Irreproducible-Discovery-Rate)'><b>Reproducibility QC and Peak Detection Statistics (Irreproducible Discovery Rate)</b>
<table border='1' style='margin-right:15px;margin-bottom:15px'>
<tr><th nowrap align=left>Nt</th><td>7280</td></tr>
<tr><th nowrap align=left>N1</th><td>8200</td></tr>
<tr><th nowrap align=left>N2</th><td>2871</td></tr>
<tr><th nowrap align=left>Np</th><td>10422</td></tr>
<tr><th nowrap align=left>N optimal</th><td>10422</td></tr>
<tr><th nowrap align=left>N conservative</th><td>7280</td></tr>
<tr><th nowrap align=left>Optimal Set</th><td>pooled_pseudo_rep</td></tr>
<tr><th nowrap align=left>Conservative Set</th><td>rep1-rep2</td></tr>
<tr><th nowrap align=left>Rescue Ratio</th><td>1.4315934065934066</td></tr>
<tr><th nowrap align=left>Self Consistency Ratio</th><td>2.856147683733891</td></tr>
<tr><th nowrap align=left>Reproducibility Test</th><td>borderline</td></tr>
</table></div><div id='help-idr'><p><ul><li>N1: Replicate 1 self-consistent IDR 0.05 peaks (comparing two pseudoreplicates generated by subsampling Rep1 reads) </li><li>N2: Replicate 2 self-consistent IDR 0.05 peaks (comparing two pseudoreplicates generated by subsampling Rep2 reads) </li><li>Nt: True Replicate consistent IDR 0.05 peaks (comparing true replicates Rep1 vs Rep2 ) </li><li>Np: Pooled-pseudoreplicate consistent IDR 0.05 peaks (comparing two pseudoreplicates generated by subsampling pooled reads from Rep1 and Rep2 ) </li><li>Self-consistency Ratio: max(N1,N2) / min (N1,N2) </li><li>Rescue Ratio: max(Np,Nt) / min (Np,Nt) </li><li>Reproducibility Test: If Self-consistency Ratio >2 AND Rescue Ratio > 2, then 'Fail' else 'Pass' </li></ul></p><br></div><figure style='display: inline-block;'><img src='./peak/spp/idr/true_reps/rep1-rep2/MITF_ANCG_Output_rep1-rep2.unthresholded-peaks.txt.png' width='800'><figcaption style='text-align: center;'><b>true reps (rep1-rep2)</b></figcaption></figure>&nbsp<figure style='display: inline-block;'><img src='./peak/spp/idr/pooled_pseudo_reps/MITF_ANCG_Output_ppr.unthresholded-peaks.txt.png' width='800'><figcaption style='text-align: center;'><b>pooled pseudo-reps</b></figcaption></figure>&nbsp<figure style='display: inline-block;'><img src='./peak/spp/idr/pseudo_reps/rep1/MITF_ANCG_Output_rep1-pr.unthresholded-peaks.txt.png' width='800'><figcaption style='text-align: center;'><b>rep1 pseudo-reps</b></figcaption></figure>&nbsp<figure style='display: inline-block;'><img src='./peak/spp/idr/pseudo_reps/rep2/MITF_ANCG_Output_rep2-pr.unthresholded-peaks.txt.png' width='800'><figcaption style='text-align: center;'><b>rep2 pseudo-reps</b></figcaption></figure>&nbsp</div><br><br></body></html>